linode/manager
# refactor: Update the `usePagination` hook to use Query Params instead of state

## Description 📝

- Updates the `usePagination` hook so it is more useful
  - Uses query params instead of state so page persists when users use browser navigation
- Updates `<SupportTicketLanding />` to be a functional component and improves its use of query params
  - I had to do this because I wanted the back button to work correctly
- Adds Pagination controls to Firewalls landing because I forgot to do that in https://github.com/linode/manager/pull/8889

## Preview 📷

### Before
Notice how when the user goes **back**, they don't return to the page they were perviously on

https://user-images.githubusercontent.com/115251059/227382914-40bf433c-fe2d-471f-9987-6af60b542ac3.mov

### After
Notice how when the user goes **back**, they **do** return to the page they were previously on

https://user-images.githubusercontent.com/115251059/227383013-0608cac0-9c68-4e93-a754-6eb3891e2d64.mov

## How to test 🧪

- Verify that the SupportTicketLanding page still works 
- Spin up at least 25 of either an Image, LKE Cluster, Database, Firewall, Volume, or any other entity that uses the `usePagination` hook
  - Verify that pagination works as expected
  - Verify that browser navigation works as expected (going back from a details page should bring you back to the previous page you were on)
  - Verify that changing the page size works and that is persists in user preferences
